test/test_magnet.cpp

```diff
@@ -224,6 +224,23 @@ TORRENT_TEST(magnet)
 	p2 = s->abort();
 }
 
+TORRENT_TEST(parse_escaped_hash_parameter)
+{
+	error_code ec;
+	add_torrent_params p;
+	parse_magnet_uri("magnet:?xt=urn%3Abtih%3Acdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd", p, ec);
+	TEST_CHECK(!ec);
+	TEST_EQUAL(aux::to_hex(p.info_hash), "cdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd");
+}
+
+TORRENT_TEST(parse_invalid_escaped_hash_parameter)
+{
+	error_code ec;
+	add_torrent_params p;
+	parse_magnet_uri("magnet:?xt=urn%%3A", p, ec);
```

Comment:
in fact, perhaps there should be a test where the "urn:btih:" part is not escaped, but a character in the info-hash is